src/main/java/teammates/logic/core/SendgridService.java

```diff
@@ -0,0 +1,45 @@
+package teammates.logic.core;
+
+import java.io.IOException;
+
+import org.jsoup.Jsoup;
+
+import com.google.appengine.labs.repackaged.org.json.JSONException;
+
+import teammates.common.util.Config;
+import teammates.common.util.EmailWrapper;
+
+public class SendgridService implements EmailSenderService {
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Sendgrid parseToEmail(EmailWrapper wrapper) throws JSONException {
+        Sendgrid email = new Sendgrid(Config.SENDGRID_USERNAME, Config.SENDGRID_PASSWORD);
+        email.setFrom(wrapper.getSenderEmail());
+        email.setFromName(wrapper.getSenderName());
+        email.setReplyTo(wrapper.getReplyTo());
+        for (String recipient : wrapper.getRecipientsList()) {
+            email.addTo(recipient);
+        }
+        if (!wrapper.getBccList().isEmpty()) {
+            // Sendgrid does not support multiple BCCs
+            email.setBcc(wrapper.getBccList().get(0));
```

Comment:
The `Sendgrid.java` that we have right now is the old way of doing stuff. The new way which is using external jar no longer has this limitation, I'll fix it altogether with adding Mailgun and Mailjet.